\subsection{Functional Requirements Review}
%Tienie
\subsubsection{Threads.getThreadStats}








%Vivian
\subsubsection{Import Thread Appraisal}
\paragraph{Data in an external file that was created using the exportThreadAppraisal function is used.\\} 
\textbf{\emph{Partial Passed}} \\
The function do make use of a file. The file is sent as arguments in the importThreadAppraisal function (parameters are directory and  fileName). So in this function they assume this is the file created by the exportThreadAppraisal function.
\par

\paragraph{The data set is associated with only one member and only one specified appraisal.\\}
\textbf{\emph{Failed}} \\
The function never actually check if the data set is about only one member and only one specific appraisal. Therefore the data cannot be deemed as eligible for import.
\par

\paragraph{A record contains all detail about the post along with a field that should contain an ordinal number that represents the levels of the specified appraisal.\\}
\textbf{\emph{Passed}} \\
The record does contain all detail about the post and a field that represents the leves of the specified appraisal which is the appraisalValue.
\par

\paragraph{Edits to the data is ignored when importing a thread appraisal.\\}
\textbf{\emph{Failed}} \\
The function does not prevent edits to the data and there is not clear indicated of ignoring such edits.
\par

\paragraph{For each record the assignAppraisalToPost function is applied.\\}
\textbf{\emph{Passed}} \\
The function does call assignAppraisalToPost for each record.
\par

\paragraph{The appraisal level as stored in the file for each post is updated as an appraisal assigned by the member associated with the data set.\\}
\textbf{\emph{Failed}} \\
The function does not use the appraisal level to update the data set.
\par

\paragraph{Check validity of member and appraisal.\\}
\textbf{\emph{Partial Passed}} \\
The function does check the validity of the member, however they have a dummy function that only returns true.\\
The function does check the validity of the appraisal, however no exception is thrown/raised it only returns true or false and there is also no means of catching an exception, that is the isValid function is not surrounded with try/catch blocks to catch exceptions and therefore the service delivery is not stopped when the appraisal is not valid.\\
The appraisal level is check for out of range, however no exception is thrown/raised when is it out of range and there the service delivery is not stopped if the appraisal level is out of range.
\par

\subsubsection{Export Thread}
\paragraph{An external file is created.\\}
\textbf{\emph{Partial Passed}} \\
The function does create an external file, however :
	\begin{itemize}
	\item{The data generated to by the queryThread function is assumed to be given as an argument to the function  (parameter threadObject).}
	\item{The data is actually not correctly stored in the file since the threadObject is not parsed as a JSON-string. The object is directly stored in the file. So the actual data of the threadObject is not stored in the file.}
	\end{itemize}
\par

\subsubsection{Import Thread}
\paragraph{An external file that was created by the exportThread function is used to restore the content of a thread.\\}
\textbf{\emph{Passed}} \\
The function does use an external file. Again the file is passed as arguments to the function (parameters directory and fileName), so they are assuming this file was generated by the exportThread function before importThread is called with the file as arguments.
\par

\paragraph{The content of the external file is used to restore content of a thread/post.\\}
\textbf{\emph{Failed}} \\
The function never tries to restore or even store the content of the external file on the Buzz Space.
\par

\paragraph{A post is added if and only if it is not in the Buzz Space already.\\}
\textbf{\emph{Failed}} \\
The function never adds the content to the Buzz Space and therefore does not check if it is already in the Buzz Space. 
\par
